src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,619 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/scenes/SceneTableImpl.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <stdlib.h>
+
+namespace chip {
+namespace scenes {
+
+enum SceneImplTLVTag
+{
+    kTagSceneCount = 1,
+    kTagNext,
+};
+
+using SceneTableEntry = DefaultSceneTableImpl::SceneTableEntry;
+using SceneStorageId  = DefaultSceneTableImpl::SceneStorageId;
+using SceneData       = DefaultSceneTableImpl::SceneData;
+
+struct FabricHavingSceneList : public CommonPersistentData::FabricList
+{
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        key = DefaultStorageKeyAllocator::SceneFabricList();
+        return CHIP_NO_ERROR;
+    }
+};
+
+static constexpr size_t kPersistentBufferMax = 256;
```

Comment:
How was this computed?

Reply:
Please find a way to express this more clearly and add a comment for why it is like this...

Reply:
Was expressed in comment in latest commit, please confirm it is clear now